Gems/ROS2/Code/Source/Manipulator/MotorizedJointComponent.cpp

```diff
@@ -0,0 +1,275 @@
+    {
+        const float measurement = ComputeMeasurement(time);
+        if (m_testSinusoidal)
+        {
+            m_setpoint = m_sinDC + m_sinAmplitude * AZ::Sin(m_sinFreq * time.GetSeconds());
+        }
+        const float control_position_error = (m_setpoint + m_zeroOffset) - measurement;
+        m_error = control_position_error; // TODO decide if we want to expose this control error.
+
+        if (m_debugDrawEntity.IsValid())
+        {
+            if (m_linear)
+            {
+                AZ::Transform transform = AZ::Transform::Identity();
+                transform.SetTranslation(m_jointDir * (m_setpoint + m_zeroOffset));
+                AZ::TransformBus::Event(
+                    m_debugDrawEntity, &AZ::TransformBus::Events::SetLocalTM, transform * m_debugDrawEntityInitialTransform);
+            }
+            else
+            {
+                AZ_Assert(false, "Not implemented");
+            }
+        }
+
+        const uint64_t deltaTimeNs = deltaTime * 1'000'000'000;
+        float speed_control = m_pidPos.ComputeCommand(control_position_error, deltaTimeNs);
+
+        if (measurement <= m_limits.first)
+        {
+            // allow only positive control
+            speed_control = AZStd::max(0.f, speed_control);
+        }
+        else if (measurement >= m_limits.second)
+        {
+            // allow only negative control
+            speed_control = AZStd::min(0.f, speed_control);
+        }
+
+        if (m_debugPrint)
+        {
+            AZ_Printf(
+                "MotorizedJointComponent",
+                " %s | pos: %f | err: %f | cntrl : %f | set : %f |",
+                GetEntity()->GetName().c_str(),
+                measurement,
+                control_position_error,
+                speed_control,
+                m_setpoint);
+        }
+        SetVelocity(speed_control, deltaTime);
+    }
+
+    float MotorizedJointComponent::ComputeMeasurement(AZ::ScriptTimePoint time)
+    {
+        AZ::Transform transform;
+        if (!m_measurementReferenceEntity.IsValid())
+        {
+            AZ::TransformBus::EventResult(transform, this->GetEntityId(), &AZ::TransformBus::Events::GetLocalTM);
+        }
+        else
+        {
+            AZ::Transform transformStepParent;
+            AZ::TransformBus::EventResult(transformStepParent, m_measurementReferenceEntity, &AZ::TransformBus::Events::GetWorldTM);
+            AZ::Transform transformStepChild;
+            AZ::TransformBus::EventResult(transformStepChild, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
+            transform = transformStepParent.GetInverse() * transformStepChild;
+        }
+        if (m_linear)
+        {
+            const float last_position = m_currentPosition;
+            m_currentPosition = transform.GetTranslation().Dot(this->m_jointDir);
+            if (m_lastMeasurementTime > 0)
```

Comment:
This variable is uninitiated and can be checked here before it is set.
It should be initialized to 0 in the class declaration in the header file